chunky/src/java/se/llbit/chunky/world/Dimension.java

```diff
@@ -202,43 +114,54 @@ public Heightmap getHeightmap() {
     return heightmap;
   }
 
-  /** Called when a new region has been discovered by the region parser. */
-  public void regionDiscovered(RegionPosition pos) {
-    synchronized (this) {
-      regionMap.computeIfAbsent(pos.getLong(), p -> createRegion(pos));
+  /**
+   * @return A user presentable name of the dimension
+   */
+  public abstract String toString();
+
+  /** Add a chunk deletion listener. */
+  public void addChunkDeletionListener(ChunkDeletionListener listener) {
+    synchronized (chunkDeletionListeners) {
```

Comment:
Could we use some concurrent collection instead of synchronizing?

Reply:
I'm not sure this needs to be synchronised at all honestly, I'll go through and verify
these listenerse were just moved from a different place